### PR TITLE
Add RAII guard for cancelling fetch when the consumer no longer cares about it

### PR DESCRIPTION
**Pull request by Manishearth**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19329)

### CODE REVIEW

**Comment by Manishearth** [comment]
r? @jdm 

We can add this to more consumers but I think it's better to make easy bugs out of this. XHR and loading are the main two consumers anyway.

File: components/script_traits/script_msg.rs [hunk 1]
```diff
@@ -71,7 +71,7 @@ pub enum LogEntry {
 pub enum ScriptMsg {
     /// Requests are sent to constellation and fetches are checked manually
     /// for cross-origin loads
-    InitiateNavigateRequest(RequestInit),
+    InitiateNavigateRequest(RequestInit, /* cancellation_chan */ Option<IpcReceiver<()>>),
```
Comment (jdm) [inline components/script_traits/script_msg.rs]: It's not clear why this is optional.

**Review by jdm — CHANGES_REQUESTED** [review]

**Pull request merged**